src/contexts/APIRouterContext/Provider.tsx

```diff
@@ -0,0 +1,157 @@
+
+      if (accountRs === undefined || handleFetchAccountIDFromRS === undefined) {
+        // TODO: error reporting this properly
+        return;
+      }
+
+      // make sure address is valid
+      if (!isValidAddress(toAddress)) {
+        return;
+      }
+
+      const recipientAccountId = await handleFetchAccountIDFromRS(toAddress);
+      const tx: IUnsignedTransaction = {
+        senderPublicKey: publicKey, // publicKey from useAccount() hook
+        senderRS: accountRs, // accountRs from useAccount() hook
+        // sender: "123", // required in some situations?
+        feeNQT: standardFee,
+        version: 1,
+        phased: false,
+        type: 0,
+        subtype: 0,
+        attachment: { "version.OrdinaryPayment": 0 },
+        amountNQT: amount, // TODO: use converter function, but for now it's nice for cheaper testing
+        recipientRS: toAddress,
+        recipient: recipientAccountId,
+        ecBlockHeight: 0, // must be included
+        deadline: standardDeadline,
+        timestamp: Math.round(Date.now() / 1000) - JUPGenesisTimestamp, // Seconds since Genesis. sets the origination time of the tx (since broadcast can happen later).
+        secretPhrase: "",
+      };
+
+      afterSecretCB.current = _handleSendJUP.bind(null, tx);
+      setRequestUserSecret(true);
+
+      return true;
+    },
+    [_handleSendJUP, accountRs, handleFetchAccountIDFromRS, publicKey]
+  );
+
+  const handleSecretEntry = useCallback((secretInput) => {
+    setUserSecretInput(secretInput);
+  }, []);
+
+  const handleCloseSeedCollection = useCallback(
+    (isSuccess: boolean) => {
+      let messageVariant: VariantType = "warning";
+      setRequestUserSecret(false);
+
+      if (isSuccess) {
+        messageVariant = "success";
+      }
+
+      enqueueSnackbar(messageText.transaction.cancel, { variant: messageVariant });
+    },
+    [enqueueSnackbar]
+  );
+
+  const handleSubmitSecret = useCallback(async () => {
+    let result;
+    try {
+      if (afterSecretCB.current !== undefined) {
+        result = await afterSecretCB.current(userSecretInput);
+      }
+    } catch (e) {
+      console.error("failed to execute api call after confirm & send", e);
+    }
+
+    console.log("result:", result);
+
+    // flush the callback to avoid future calling of it unintentionally
+    // flush the user's seedPhrase for security
+    afterSecretCB.current = undefined;
+    setUserSecretInput("");
+
+    // close seed collection dialog without firing the closeFn (prevents a duplicate notification)
+    setRequestUserSecret(false);
+  }, [userSecretInput]);
+
+  return (
+    <Context.Provider
+      value={{
+        sendJUP: handleSendJUP,
+      }}
+    >
+      {children}
+      {/* dialog handles obtaining secret phrase if needed by the current action */}
+      <JUPDialog isOpen={requestUserSecret} closeFn={() => handleCloseSeedCollection(false)}>
+        <DialogContent>
+          <Box sx={{ minWidth: "600px", height: "300px" }}>
```

Comment:
is this styling needed anymore with the way the JUPDialog has been refactored to have a defined size?

Reply:
we'll have to test it. i don't think it's necessary anymore, no, but there maybe an incident where it's nice to have.

check it when the screen is very narrow (mobile sizes) and ensure that it's working ok.

Reply:
the fixed height should probably go anyways, that's not a very good idea imo. a min-height would be a decent idea however.